### ISSUE TITLE
Featured competitions

### ISSUE THREAD

**Issue by madclam** [codalab/codalab-competitions#2816]
https://competitions.codalab.org/

We need to check the formula for "featured competitions" because they are supposed to be drawn from recent competitions and they as seem to be from 2017.

**Comment by Didayolo** [comment]
More information in this old issue: #2454

**Comment by Didayolo** [comment]
Sometimes (~ 1 times out of 5) when pushing to the project, the following CircleCI test is failing:

```
-- Docs: https://docs.pytest.org/en/latest/warnings.html
=========================== short test summary info ============================
FAILED apps/common/tests/test_featured_competitions.py::FeaturedCompetitionsTests::test_get_featured_competitions_returns_newer_competitions
================= 1 failed, 261 passed, 730 warnings in 34.39s =================

Exited with code exit status 1

```

If the test is well designed (this need to be check) then it reveals that the featured competitions are not always newer competitions.

**Comment by Didayolo** [comment]
The function to update is:

https://github.com/codalab/codalab-competitions/blob/177da908e02228233abd98e96bdfd9aede966fcd/codalab/apps/common/competition_utils.py#L47

**Comment by Didayolo** [comment]
The failing test in question:

```
E       AssertionError: assert <Competition: competition_old> not in [<Competition: competition_old>]
E        +  where <Competition: competition_old> = <apps.common.tests.test_featured_competitions.FeaturedCompetitionsTests testMethod=test_get_featured_competitions_returns_newer_competitions>.competition_old
```

**Comment by Didayolo** [comment]
Fixed by #3308
